inter/proposal_test.go

```diff
@@ -0,0 +1,193 @@
+package inter
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"math/big"
+	"testing"
+
+	"github.com/0xsoniclabs/sonic/inter/pb"
+	"github.com/Fantom-foundation/lachesis-base/hash"
+	"github.com/Fantom-foundation/lachesis-base/inter/idx"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestProposal_Hash_IsShaOfFieldConcatenation(t *testing.T) {
+
+	// The procedure of computing the hash of a proposal is critical for
+	// the consensus protocol. It is important to ensure that the hash is
+	// computed correctly and consistently. Thus, this test provides a
+	// second implementation of the hash function. If you have to change this
+	// test, make sure you understand the implications on consensus.
+
+	for i := range 5 {
+
+		proposal := &Proposal{
+			Number:     idx.Block(1 + i),
+			Attempt:    uint32(2 + i),
+			ParentHash: [32]byte{0: 1, 1: byte(i), 31: 2},
+			Time:       Timestamp(3 + i),
+			Randao:     [32]byte{0: 3, 1: byte(i), 21: 4},
+			Transactions: []*types.Transaction{
+				types.NewTx(&types.LegacyTx{Nonce: 1}),
+				types.NewTx(&types.LegacyTx{Nonce: 2}),
+				types.NewTx(&types.LegacyTx{Nonce: uint64(i)}),
+			},
+		}
+
+		hash := func(proposal *Proposal) hash.Hash {
```

Comment:
does this closure need to be inside the loop or could it be moved before it ? it would also prevent `proposal` shadowing.